src/core/Logger.cpp

```diff
@@ -20,48 +29,50 @@ Logger::Logger() : m_buf(), m_severity(LSEVERITY_INFO), m_output(&m_buf), m_colo
 {
 }
 
-/* Reset code */
-static const char* ANSI_RESET = "\x1b[0m";
-
-/* Normal Colors */
-static const char* ANSI_RED = "\x1b[31m";
-static const char* ANSI_GREEN = "\x1b[32m";
-static const char* ANSI_ORANGE = "\x1b[33m";
-static const char* ANSI_YELLOW = "\x1b[33;2m";
-//static const char* ANSI_BLUE = "\x1b[34m";
-//static const char* ANSI_CYAN = "\x1b[36m";
-static const char* ANSI_GREY = "\x1b[37m";
-
-/* Bold Colors */
-//static const char* ANSI_BOLD_RED = "\x1b[31;1m";
-//static const char* ANSI_BOLD_GREEN = "\x1b[32:1m";
-//static const char* ANSI_BOLD_YELLOW = "\x1b[33:1m";
-//static const char* ANSI_BOLD_WHITE = "\x1b[37;1m";
-
-/* Dark Colors */
-//static const char* ANSI_DARK_RED = "\x1b[31;2m";
-//static const char* ANSI_DARK_GREEN = "\x1b[32;2m";
-static const char* ANSI_DARK_CYAN = "\x1b[36;2m";
-static const char* ANSI_DARK_GREY = "\x1b[37;2m";
+#ifdef _WIN32
+#define STDOUT GetStdHandle(STD_OUTPUT_HANDLE)
+#define RESET (SetConsoleTextAttribute(STDOUT, old_colors.wAttributes) ? "" : "")
+
+#define RED (SetConsoleTextAttribute(STDOUT, FOREGROUND_RED) ? "" : "")
```

Comment:
As much as I dislike preprocessor macros, you should at least macro-ify the Windows' call so the code isn't ugly, i.e.:

```
#define WINCOLOR(x) (SetConsoleTextAttribute(STDOUT, x) ? "" : "")

#define RED WINCOLOR(FOREGROUND_RED)
#define GREEN WINCOLOR(FOREGROUND_GREEN)
```

etc...